### PR TITLE
[DOCS] add `public_api` decorator to `TableAsset` and `QueryAsset`

### PR DESCRIPTION
**Pull request by Kilo59**
Add `@public_api` decorator to `SQLDatasource`, `TableAsset`, and `QueryAsset` so that they show up in our API docs.

### Note
Due to the way sphinx is dealing with inheritance most of the important methods on these assets will likely not show up.
* `.add_splitter_datetime_part()`
* `.add_splitter_column_value()`
* `.add_splitter_year_and_month()`
*  etc.

~We would need to use the `:inherited-members:` directive.~
cc @anthonyburdi 

https://stackoverflow.com/questions/30344289/how-can-i-show-inherited-members-of-a-class-in-my-sphinx-documentation

---------

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/datasource/fluent/sql_datasource.py [hunk 1]
```diff
@@ -762,6 +771,7 @@ def as_selectable(self) -> sqlalchemy.Selectable:
         raise NotImplementedError
 
 
+@public_api
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py:774]: These assets are not dynamically defined, they just didn't have the `@public_api` decorator and so weren't showing up in our API docs.

But also I should note that users don't ever need to initiate these classes directly, and our docs should never tell users to do that. 
We have factory methods for instantiating them.

These classes are somewhat implementation details that users shouldn't have to know about.

Comment (anthonyburdi) [reply great_expectations/datasource/fluent/sql_datasource.py:774]: Would it make sense to change the class docstring to point users to the factory method? So if they end up on the API docs page they will see the recommended way to do things.

**Comment by anthonyburdi** [comment]
@Kilo59 FYI We do use the `:inherited-members:` directive when building API docs for both classes and modules:
- https://github.com/great-expectations/great_expectations/blob/develop/docs/sphinx_api_docs_source/build_sphinx_api_docs.py#L511
- https://github.com/great-expectations/great_expectations/blob/develop/docs/sphinx_api_docs_source/build_sphinx_api_docs.py#L536

**Comment by anthonyburdi** [comment]
FYI I kicked off a deploy preview manually to see what this looks like when rendered in netlify.

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thanks for adding these.

**Pull request merged**